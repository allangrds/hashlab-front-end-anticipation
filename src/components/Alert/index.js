import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './styles.module.css'

const Alert = ({
  className,
  text,
  type,
}) => (
  (
    <div className={classNames(
      className,
      styles.alert,
      styles[type],
    )}
    >
      { text }
    </div>
  )
)

Alert.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
  type: PropTypes.oneOf([
    'info',
    'warning',
    'error',
  ]),
}

Alert.defaultProps = {
  className: null,
  text: null,
  type: null,
}

export default Alert
