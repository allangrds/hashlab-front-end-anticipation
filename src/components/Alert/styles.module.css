@import "../../assets/css/index.css";

.alert {
  border-radius: 4px;
  font-family: "Source Sans Pro", sans-serif;
  font-size: 1em;
  height: 58px;
  padding: 20px 12px;
}

.info {
  color: var(--color-white);
  background-color: var(--color-blue-2);
}

.warning {
  color: var(--color-white);
  background-color: var(--color-orange-1);
}

.error {
  color: var(--color-white);
  background-color: var(--color-error);
}
