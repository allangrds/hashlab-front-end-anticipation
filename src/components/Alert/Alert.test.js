import React from 'react'
import { render } from '@testing-library/react'

import Alert from '.'

test('Alert info should render correctly', () => {
  const text = 'Olá mundo'

  const {
    container,
    getByRole,
  } = render(
    <Alert
      type="info"
      text={text}
    />,
  )

  expect(getByRole('alert')).toHaveTextContent(text)
  expect(container).toMatchSnapshot()
})

test('Alert warning should render correctly', () => {
  const text = 'Olá mundo'

  const {
    container,
    getByRole,
  } = render(
    <Alert
      type="warning"
      text={text}
    />,
  )

  expect(getByRole('alert')).toHaveTextContent(text)
  expect(container).toMatchSnapshot()
})

test('Alert error should render correctly', () => {
  const text = 'Olá mundo'

  const {
    container,
    getByRole,
  } = render(
    <Alert
      type="error"
      text={text}
    />,
  )

  expect(getByRole('alert')).toHaveTextContent(text)
  expect(container).toMatchSnapshot()
})
