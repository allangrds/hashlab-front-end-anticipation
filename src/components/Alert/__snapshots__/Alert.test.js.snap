// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert error should render correctly 1`] = `
<div>
  <div
    class="alert error"
    role="alert"
  >
    Olá mundo
  </div>
</div>
`;

exports[`Alert info should render correctly 1`] = `
<div>
  <div
    class="alert info"
    role="alert"
  >
    Olá mundo
  </div>
</div>
`;

exports[`Alert warning should render correctly 1`] = `
<div>
  <div
    class="alert warning"
    role="alert"
  >
    Olá mundo
  </div>
</div>
`;
